### PR TITLE
feat: CustomEventType placeholder and API fixes

### PR DESCRIPTION
**Pull request by mihaic195**
- Feat: Text/Multiline Text Custom Inputs can have (optional) placeholders to display on booking page
- Fix: if custom inputs existed on a specific event type, the event type cannot be deleted due to DB association

Screenshot 2021-08-13 at 17 39 26
Screenshot 2021-08-13 at 17 39 41
Screenshot 2021-08-13 at 17 40 22
Screenshot 2021-08-13 at 17 40 48

### CODE REVIEW

File: pages/event-types/[type].tsx [hunk 1]
```diff
@@ -1008,6 +1017,23 @@ export default function EventTypePage({
...
+                      <label htmlFor="placeholder" className="block text-sm font-medium text-gray-700">
+                        Placeholder
+                      </label>
+                      <div className="mt-1">
+                        <input
+                          type="text"
+                          name="placeholder"
+                          id="placeholder"
+                          className="shadow-sm focus:ring-primary-500 focus:border-primary-500 block w-full sm:text-sm border-gray-300 rounded-sm"
+                          defaultValue={selectedCustomInput?.label}
```
Comment (emrysal) [inline pages/event-types/[type].tsx:1032]: See attached video, this does not seem right to me.

**Review by emrysal — CHANGES_REQUESTED** [review]
<div class="loom-thumb-embed"><p><b>My beautiful event type | Event Type | Calendso</b> - <a target="_blank" href="https://www.loom.com/share/0048e55731ca4c92b30cbddd2656e523">Watch Video</a></p><a class="lo-gmail-video-thumb" href="https://www.loom.com/share/0048e55731ca4c92b30cbddd2656e523?utm_medium=gif" target="_blank"></a></div>

**Comment by mihaic195** [comment]
@emrysal thanks for the code review. I'll update it as soon as your PR https://github.com/calendso/calendso/pull/450 is merged...

**Comment by baileypumfleet** [comment]
The PR #450 has been merged now. Once you've fixed the conflicts and updated your PR, we can get going :)

File: pages/event-types/[type].tsx [hunk 2]
```diff
@@ -1015,6 +1024,23 @@ export default function EventTypePage({
                       />
                     </div>
                   </div>
+                  {(selectedInputOption.value === EventTypeCustomInputType.TEXT ||
+                    selectedInputOption.value == EventTypeCustomInputType.TEXTLONG) && (
```
Comment (emrysal) [inline pages/event-types/[type].tsx:1028]: ===

**Review by emrysal — APPROVED** [review]
Minor tweak, otherwise good, approved

File: pages/event-types/[type].tsx (hunk 2 above)
Comment (mihaic195) [reply pages/event-types/[type].tsx:1028]: Nice one! 😁

**Pull request merged**